src/figuresClasses.ts

```diff
@@ -1,11 +1,77 @@
-export interface Figure {}
+type Shapes = 'triangle' | 'rectangle' | 'circle';
+type Colors = 'red' | 'blue' | 'green';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shapes;
+  color: Colors;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: Shapes = 'triangle';
+
+  constructor(
+    public color: Colors,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('This figure does not have a negative value');
+    }
+
+    if (a + b <= c || b + c <= a || c + a <= b) {
+      throw new Error('This is not a triangle');
+    }
+  }
+
+  getArea(): number {
+    const p = (this.a + this.b + this.c) / 2;
+    const area = Math.sqrt(p * (p - this.a) * (p - this.b) * (p - this.c));
+
+    return Math.floor(area * 100) / 100;
+  }
+}
+
+export class Circle implements Figure {
+  shape: Shapes = 'circle';
 
-export class Circle implements Figure {}
+  constructor(
+    public color: Colors,
+    public radius: number,
+  ) {
+    if (radius <= 0) {
+      throw new Error('This figure does not have a negative value');
+    }
+  }
 
-export class Rectangle implements Figure {}
+  getArea(): number {
+    const area = Math.PI * this.radius ** 2;
+
+    return Math.floor(area * 100) / 100;
+  }
+}
+
+export class Rectangle implements Figure {
+  shape: Shapes = 'rectangle';
+
+  constructor(
+    public color: Colors,
+    public width: number,
+    public height: number,
+  ) {
+    if (width <= 0 || height <= 0) {
+      throw new Error('This figure does not have a negative value');
```

Comment:
The error message 'This figure does not have a negative value' is misleading. It would be clearer to specify the exact condition, such as 'Width and height must be positive'.